File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        write_index = 0

        for num in nums[1:]:
            if num != nums[write_index]:
                write_index += 1
                nums[write_index] = num
        
        return write_index + 1

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    if( nums.length < 2 )
        return nums.length;
    
    let i = 1, ri = 1;
    
    while( i < nums.length ) {
        if( nums[i] !== nums[i-1] ) {
            nums[ri] = nums[i];
            ri++;
        }
        
        i++;
    }
    
    return ri;
};
